SAT/SATLiteral.hpp

```diff
@@ -18,77 +18,49 @@
 
 #include <ostream>
 
-#include "Shell/Options.hpp"
 #include "Debug/Assertion.hpp"
-#include "Kernel/Clause.hpp"
+#include "Lib/Hash.hpp"
 
 namespace SAT {
 
-using namespace Kernel;
-
 class SATLiteral
 {
 public:
-  inline SATLiteral() {}
-  explicit inline SATLiteral(unsigned content) :_content(content) {}
+  SATLiteral() = default;
+
   /**
    * Create a SAT literal of variable @b var and polarity &b polarity
    *
-   * @b var must be greater than 0 and @b polarity either 1 or 0 (for positive or negative)
+   * @b var must be greater than 0
    */
-  inline SATLiteral(unsigned var, unsigned polarity) :_polarity(polarity), _var(var)
-  { ASS_G(var,0); ASS_NEQ(var,0x7FFFFFFF); }
-
-
-  inline void set(unsigned var, bool positive) {
-    _var=var;
-    _polarity=positive?1:0;
-  }
-  inline void setContent(unsigned content) {
-    _content=content;
-  }
-
-  inline bool isPositive() const { return _polarity; }
-  inline bool isNegative() const { return !_polarity; }
-  inline unsigned var() const { return _var; }
-  inline unsigned polarity() const { return _polarity; }
-  inline unsigned oppositePolarity() const { return 1-_polarity; }
-  inline unsigned content() const { return _content; }
+  SATLiteral(unsigned var, bool polarity)
+  : _content(polarity ? var : -int(var))
```

Comment:
Would it make sense to assert that `var < 2^31`?

Reply:
Yes, I think so. Or whatever the fancy C++ way of checking `unsigned` fits in `int` is. I'll look it up.